optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverManagerConfig.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    public SolverConfig getSolverConfig() {
+        return solverConfig;
+    }
+
+    public void setSolverConfig(SolverConfig solverConfig) {
+        this.solverConfig = solverConfig;
+    }
+
+    public String getParallelSolverCount() {
+        return parallelSolverCount;
+    }
+
+    public void setParallelSolverCount(String parallelSolverCount) {
+        this.parallelSolverCount = parallelSolverCount;
+    }
+
+    public Class<? extends ThreadFactory> getThreadFactoryClass() {
+        return threadFactoryClass;
+    }
+
+    public void setThreadFactoryClass(Class<? extends ThreadFactory> threadFactoryClass) {
+        this.threadFactoryClass = threadFactoryClass;
+    }
+
+    // ************************************************************************
+    // With methods
+    // ************************************************************************
+
+    public SolverManagerConfig withParallelSolverCount(String parallelSolverCount) {
+        this.parallelSolverCount = parallelSolverCount;
+        return this;
+    }
+
+    public SolverManagerConfig withThreadFactoryClass(Class<? extends ThreadFactory> threadFactoryClass) {
+        this.threadFactoryClass = threadFactoryClass;
+        return this;
+    }
+
+    // ************************************************************************
+    // Builder methods
+    // ************************************************************************
+
+    public Integer resolveParallelSolverCount() {
+        int availableProcessorCount = getAvailableProcessors();
+        Integer resolvedParallelSolverCount;
+        if (parallelSolverCount == null || parallelSolverCount.equals(PARALLEL_SOLVER_COUNT_AUTO)) {
+            // Leave one for the Operating System and 1 for the solver thread, take the rest
+            resolvedParallelSolverCount = resolveParallelSolverCountAutomatically(availableProcessorCount);
+        } else {
+            resolvedParallelSolverCount = ConfigUtils.resolveThreadPoolSizeScript(
+                    "parallelSolverCount", parallelSolverCount, PARALLEL_SOLVER_COUNT_AUTO);
+        }
+        if (resolvedParallelSolverCount < 1) {
+            throw new IllegalArgumentException("The parallelSolverCount (" + parallelSolverCount
+                    + ") resulted in a resolvedParallelSolverCount (" + resolvedParallelSolverCount
+                    + ") that is lower than 1.");
+        }
+        if (resolvedParallelSolverCount > availableProcessorCount) {
+            logger.warn("The resolvedParallelSolverCount ({}) is higher "
+                            + "than the availableProcessorCount ({}), which is counter-efficient.",
+                    resolvedParallelSolverCount, availableProcessorCount);
+            // Still allow it, to reproduce issues of a high-end server machine on a low-end developer machine
+        }
+        return resolvedParallelSolverCount;
+    }
+
+    protected int getAvailableProcessors() {
```

Comment:
Should this method be made static?

Reply:
Also, it's a simple wrapper for another method. Why not remove it altogether?

Reply:
No, because some unit tests override it (at least for SolverConfig, but they'll do so here too) to mimic computers with a different number of cpu's than the random jenkins job that runs these tests too.

Reply:
Ah, makes sense.